### PR TITLE
NEXUS-21749 Resolve technical debt in Helm

### PR DESCRIPTION
**Pull request by proteus1121**
https://issues.sonatype.org/browse/NEXUS-21749

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public UploadResponse handle(Repository repository, ComponentUpload upload) thro
 
     String fileName = payload.getName() != null ? payload.getName() : StringUtils.EMPTY;
 
-    try (TempBlob tempBlob = storageFacet.createTempBlob(payload, HASH_ALGORITHMS)) {
+    try (TempBlob tempBlob = storageFacet.createTempBlob(payload, HelmFormat.HASH_ALGORITHMS)) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java]: static import

Comment (proteus1121) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java]: Fixed

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java [hunk 2]
```diff
@@ -12,105 +12,91 @@
...
-  private String version;
-
-  public String getDescription() {
-    return description;
-  }
-
-  public void setDescription(final String description) {
-    this.description = description;
-  }
+  private Map<HelmProperties, Object> attributesEnumMap;
```
[the remark this answers is not in the document]
Comment (proteus1121) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java]: Done

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java [hunk 3]
```diff
@@ -12,105 +12,91 @@
...
   }
 
-  public void setEngine(final String engine) {
-    this.engine = engine;
+  public HelmAttributes(final Map<String, Object> attributesMap) {
+    attributesEnumMap = new EnumMap<>(HelmProperties.class);
+    attributesMap.forEach((key, value) -> {
+      Optional<HelmProperties> propertyOpt = HelmProperties.findByPropertyName(key);
+      if (value != null && propertyOpt.isPresent()) {
+        attributesEnumMap.put(propertyOpt.get(), value);
```
Comment (mlukaretkyi) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java]: HelmProperties.findByPropertyName always return optional so in this place, we do not need to check for null
You can replace your code
`    HelmProperties.findByPropertyName(key)
          .ifPresent(properties -> attributesEnumMap.put(properties, value));`

Comment (proteus1121) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java]: value which we get from file could be null

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java [hunk 4]
```diff
@@ -37,73 +58,151 @@
...
+  }
+
+  /**
+   * Save an asset and create blob.
+   *
+   * @return blob content
+   */
+  @Override
+  public Content saveAsset(final StorageTx tx,
+                            final Asset asset,
```
Comment (mlukaretkyi) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java]: Method can return null. please add @Nullable

Comment (proteus1121) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/HelmFacetImpl.java]: Done

**Pull request merged**